build.sbt

```diff
@@ -16,5 +16,6 @@ parallelExecution in Test := false
 
 libraryDependencies ++= Seq(
   "org.scala-lang.modules" %% "scala-java8-compat" % "0.8.0",
-  "com.novocode" % "junit-interface" % "0.11" % "test"
+  "com.novocode" % "junit-interface" % "0.11" % "test",
+  "com.storm-enroute" %% "scalameter" % "0.8.2"
```

Comment:
Based on my experience `scalameter` is really good in case you want to store the history of runs and maybe run regression testing. It's also good as long as value you're measuring isn't less than a millisecond.
Also, I've found JVM to be a better tool in answering the `why?` question, as it can produce both the heatmap and the actual generated assembly. 
I think that for the kind of benchmarking that is done here now both would do fine.